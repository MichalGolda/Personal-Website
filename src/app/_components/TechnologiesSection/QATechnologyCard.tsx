import TechnologyCard from "./TechnologyCard";
import { EslintIcon, PrettierIcon } from "./TechnologyIcon";

const Icons = () => {
  return (
    <>
      <PrettierIcon />
      <EslintIcon />
    </>
  );
};

const QATechnologyCard = () => {
  return (
    <TechnologyCard
      title="QA"
      content={() => (
        <>
          Jakość kodu i finalnego projektu zapewniam dzięki:{" "}
          <b className="text-primary">Clean Code</b>,{" "}
          <b className="text-primary">KISS</b>,{" "}
          <b className="text-primary">DRY</b>,{" "}
          <b className="text-primary">YAGNI</b>,{" "}
          <b className="text-primary">Testy jednostkowe</b>,{" "}
          <b className="text-primary">Testy integracyjne</b>,{" "}
          <b className="text-primary">prettier</b>,{" "}
          <b className="text-primary">eslint</b>.
        </>
      )}
      icons={<Icons />}
    />
  );
};

export default QATechnologyCard;
