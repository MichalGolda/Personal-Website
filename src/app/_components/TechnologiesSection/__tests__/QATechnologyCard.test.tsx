import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import QATechnologyCard from "../QATechnologyCard";

describe("QATechnologyCard component", () => {
  it("renders unchanged", () => {
    const { container } = render(<QATechnologyCard />);

    expect(container).toMatchSnapshot();
  });
});
