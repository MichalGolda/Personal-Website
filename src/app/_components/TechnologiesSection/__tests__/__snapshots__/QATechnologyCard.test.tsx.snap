// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QATechnologyCard component renders unchanged 1`] = `
<div>
  <div
    class="w-full flex flex-col gap-y-4"
  >
    <h4
      class="text-lg md:text-xl lg:text-2xl text-secondary font-bold"
    >
      QA
    </h4>
    <p
      class="text-sm md:text-base text-body"
    >
      Jakość kodu i finalnego projektu zapewniam dzięki:
       
      <b
        class="text-primary"
      >
        Clean Code
      </b>
      ,
       
      <b
        class="text-primary"
      >
        KISS
      </b>
      ,
       
      <b
        class="text-primary"
      >
        DRY
      </b>
      ,
       
      <b
        class="text-primary"
      >
        YAGNI
      </b>
      ,
       
      <b
        class="text-primary"
      >
        Testy jednostkowe
      </b>
      ,
       
      <b
        class="text-primary"
      >
        Testy integracyjne
      </b>
      ,
       
      <b
        class="text-primary"
      >
        prettier
      </b>
      ,
       
      <b
        class="text-primary"
      >
        eslint
      </b>
      .
    </p>
    <div
      class="flex flex-row flex-wrap gap-4"
    >
      <img
        alt="prettier"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Fprettier.webp&w=64&q=75"
        srcset="/_next/image?url=%2Fprettier.webp&w=32&q=75 1x, /_next/image?url=%2Fprettier.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
      <img
        alt="eslint"
        class="max-w-8 max-h-8"
        data-nimg="1"
        decoding="async"
        height="32"
        loading="lazy"
        src="/_next/image?url=%2Feslint.webp&w=64&q=75"
        srcset="/_next/image?url=%2Feslint.webp&w=32&q=75 1x, /_next/image?url=%2Feslint.webp&w=64&q=75 2x"
        style="color: transparent;"
        width="32"
      />
    </div>
  </div>
</div>
`;
